llvm/lib/CodeGen/EarlyIfConversion.cpp

```diff
@@ -690,6 +717,13 @@ void SSAIfConv::convertIf(SmallVectorImpl<MachineBasicBlock *> &RemoveBlocks,
   else
     ++NumDiamondsConv;
 
+  // If both blocks are going to be merged into Head, remove "killed" flag from
+  // registers, which are killed in TBB and FBB. Otherwise, register will be
+  // killed twice in Head after splice. Double killed register is an incorrect
+  // MIR.
+  if (TBB != Tail && FBB != Tail)
+    clearDoubleKillFlags(TBB, FBB);
```

Comment:
IIUC you need to remove kill flags from `TBB` as instructions from that block are spliced _before_ those in `FBB`.

Reply:
I like your idea more than my original one.
Implementation is updated to conform your proposal.

Done.